src/html5_parser/parser/document.rs

```diff
@@ -304,6 +311,12 @@ impl PartialEq for DocumentHandle {
     }
 }
 
+impl Clone for DocumentHandle {
+    fn clone(&self) -> DocumentHandle {
+        DocumentHandle(Rc::clone(&self.0))
+    }
+}
```

Comment:
I like keeping `Document::clone(...)`, copying `Rc::clone(&reference)` instead of `reference.clone()`, as it communicates to the developer that we're only doing a fast, shallow clone of a reference.

Reply:
I also prefer that as well, but the `#[derive(Clone)]` tags (or whatever they're called) was complaining that it wasn't implemented

Reply:
Ah, gotcha.  Maybe a comment above the trait `impl` describing the preferred usage?

Reply:
Can do!

Reply:
Comment added